lib/ansible/module_utils/facts.py

```diff
@@ -3169,8 +3169,42 @@ def populate(self):
         return self.facts
 
     def get_virtual_facts(self):
-        self.facts['virtualization_type'] = ''
-        self.facts['virtualization_role'] = ''
+        rc, out, err = self.module.run_command("/usr/sbin/sysctl -n hw.product")
```

Comment:
I would use module.get_bin_path function instead of hardcoding path (future proof), same below